src/pages/apps/components/AppsAdminActionGroup.tsx

```diff
@@ -0,0 +1,73 @@
+import {Autocomplete, Grid, Paper, Stack, TextField} from '@mui/material';
+import CreateUpdateGroup from '../../groups/CreateUpdate';
+import {OktaUser, App, AppGroup} from '../../../api/apiSchemas';
+import {renderUserOption} from '../../../components/TableTopBar';
+import {displayUserName, sortGroupMemberRecords, sortGroupMembers} from '../../../helpers';
+import React from 'react';
+
+interface AppsAdminActionGroupProps {
+  currentUser: OktaUser;
+  app: App;
+  onSearchSubmit?: (appGroup: AppGroup[]) => void;
+}
+
+export const AppsAdminActionGroup: React.FC<AppsAdminActionGroupProps> = ({currentUser, app, onSearchSubmit}) => {
+  const allMembers: Record<string, OktaUser> = {};
+  const memberGroups: Record<string, AppGroup[]> = {};
+  [app.active_non_owner_app_groups].forEach((appGroupList) => {
+    (appGroupList ?? []).forEach((appGroup) => {
+      [appGroup.active_user_ownerships, appGroup.active_user_memberships].forEach((memberList) => {
+        (memberList ?? []).forEach((member) => {
+          const activeUser = member.active_user;
+          if (activeUser) {
+            allMembers[activeUser.id] = activeUser;
+            const groups = (memberGroups[activeUser.email.toLowerCase()] ||= []);
+            if (
+              !groups.find((g) => {
+                return g.id === appGroup.id;
+              })
+            ) {
+              groups.push(appGroup);
+            }
+          }
+        });
+      });
+    });
+  });
+
+  const handleSearchSubmit = (_: unknown, newValue: string | null) => {
+    const email = newValue?.split(';')[1] ?? '';
+    const appGroups = memberGroups[email] ?? app.active_non_owner_app_groups;
+    if (!!onSearchSubmit) {
+      onSearchSubmit(appGroups);
+    }
+  };
+
+  return (
+    <Grid item xs={12} className={'app-detail app-detail-admin-action-group'}>
+      <Paper
+        sx={{
+          p: 2,
+          display: 'flex',
+          alignItems: 'center',
+        }}>
+        <Stack direction="row" width="100%" justifyContent="space-between">
+          <CreateUpdateGroup defaultGroupType={'app_group'} currentUser={currentUser} app={app} />
+          <Autocomplete
```

Comment:
Looks like when selecting an option it keeps the entire encoded option data with the semi-colon. I believe there's a way to strip this out after a selection is made. During `onChange`?

![Screenshot 2024-11-04 at 5 58 37 PM](https://github.com/user-attachments/assets/1c999864-7f8d-4e0b-8d5f-d6db1b12d328)

Also I wonder if it feels more like a search box if we add the `freeSolo` prop